kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/ScheduleStrategy.java

```diff
@@ -270,4 +275,22 @@ public void onMessageConfirmed(int messageId, String topic) {
         updateState(State::onMessageEvent);
     }
 
+    @Override
+    public void onPublish() {
+        DataMessage tempDataMessage = this.connectionManager.getNextMessage();
```

Comment:
Maybe we can pass the same parameters that are provided to DataService.publish() to this method so that we can get the message priority from there sparing a db lookup.